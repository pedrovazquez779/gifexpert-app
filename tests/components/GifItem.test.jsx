import {GifItem} from '../../src/components/index.js';
import {render, screen} from '@testing-library/react';

describe('Testing <GifItem /> component', () => {

    const title = 'Goku';
    const url = 'https://goku.com/goku.jpg';

    test('Snapshot test match', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    })

    test('Should show image with the correct URL and ALT', () => {
        render(<GifItem title={title} url={url}/>);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Should show the title', () => {
        render(<GifItem title={title} url={url}/>);

        expect(screen.getByText(title)).toBeTruthy();
    })
});