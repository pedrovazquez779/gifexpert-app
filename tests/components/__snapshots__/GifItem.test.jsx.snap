// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> component Snapshot test match 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku.com/goku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
